aws/service.go

```diff
@@ -1791,6 +1794,28 @@ func (r ConnectionErrRetryer) ShouldRetry(req *request.Request) bool {
 			return true
 		}
 	}
+	var awsErr awserr.Error
```

Comment:
Why is this separate from the `req.Error != nil` above? They should be combined in some way?